rdkit/ML/Cluster/Butina.py

```diff
@@ -55,14 +55,15 @@ def ClusterData(data, nPts, distThresh, isDistData=False, distFunc=EuclideanDist
            ...
          )
          The first element for each cluster is its centroid.
-
   """
   if isDistData and len(data) > (nPts * (nPts - 1) / 2):
     logger.warning("Distance matrix is too long")
-  nbrLists = [None] * nPts
-  for i in range(nPts):
-    nbrLists[i] = []
 
+  # Create an empty matrix of booleans with a point for each i,j pair of molecule numbers.
+  # Also tally the count of all pairs, skipping duplicates, and all self-comparison.
+  # For efficiency, we never change the size of this bit bit-table.  We just flip bits.
+  matrix_passing_threshold: numpy.array = numpy.zeros([nPts, nPts], dtype=bool)
```

Comment:
The RDKit doesn't use type hints. Please remove them

Reply:
I'll update to camelCase too.